aws/src/main/java/org/apache/iceberg/aws/dynamodb/DynamoDbTableOperations.java

```diff
@@ -116,20 +119,30 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
                   .build());
       checkMetadataLocation(table, base);
       Map<String, String> properties = prepareProperties(table, newMetadataLocation);
-      persistTable(tableKey, table, properties);
+      persistTable(tableKey, table, properties, retryDetector);
       commitStatus = CommitStatus.SUCCESS;
-    } catch (ConditionalCheckFailedException e) {
-      throw new CommitFailedException(
-          e, "Cannot commit %s: concurrent update detected", tableName());
     } catch (CommitFailedException e) {
       // any explicit commit failures are passed up and out to the retry handler
       throw e;
     } catch (RuntimeException persistFailure) {
-      LOG.error(
-          "Confirming if commit to {} indeed failed to persist, attempting to reconnect and check.",
-          fullTableName,
-          persistFailure);
-      commitStatus = checkCommitStatus(newMetadataLocation, metadata);
+      boolean isConditionCheckFailedEx = persistFailure instanceof ConditionalCheckFailedException;
```

Comment:
nit: similar style comment, can avoid `is` and name it shorter, like `conditionCheckFailed`